File: ios/ios/view/PrayerTimesView.swift
//
//  PrayerTimesView.swift
//  ios
//
//  Created by Ahmed El-Helw on 8/30/21.
//  Copyright © 2021 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import shared

struct PrayerTimesView: View {
    var salahTimes: SalahTimes

    var body: some View {
        VStack {
            Text(salahTimes.name)
            Text(salahTimes.fajr)
            Text(salahTimes.dhuhr)
            Text(salahTimes.asr)
            Text(salahTimes.maghrib)
            Text(salahTimes.isha)
        }
    }
}

struct PrayerTimesView_Previews: PreviewProvider {
    static var previews: some View {
        let salahTimes = SalahTimes(name: "Makkah", fajr: "3:59 am", dhuhr: "12:20pm", asr: "3:41pm", maghrib: "7:10pm", isha: "8:36pm")
        PrayerTimesView(salahTimes: salahTimes)
    }
}

File: renderer/src/commonMain/kotlin/app/salah/view/PrayerTimesView.kt
package app.salah.view

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import app.salah.model.SalahTimes

@Composable
internal fun PrayerTimesView(salahTimes: SalahTimes) {
  Column(modifier = Modifier.padding(16.dp)) {
    Text(salahTimes.name)
    Text(salahTimes.fajr)
    Text(salahTimes.dhuhr)
    Text(salahTimes.asr)
    Text(salahTimes.maghrib)
    Text(salahTimes.isha)
  }
}